src/main/kotlin/org/jitsi/utils/TemplatedUrl.kt

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright @ 2025 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.utils
+
+import java.net.URI
+
+class TemplatedUrl(
+    private val template: String,
+    private val map: MutableMap<String, String> = mutableMapOf(),
```

Comment:
If someone passes in a MutableMap they can then manipulate it outside of this object, which feels like a bad idea.  Make the argument a map, and copy it to a private mutableMap?